### ISSUE TITLE
0.10.2 Regression: Windows installer fails due to malformatted tar

### ISSUE THREAD

**Issue by nickbabcock** [rustwasm/wasm-pack#1097]
## 🐛 Bug description

```bash
curl -L "https://github.com/rustwasm/wasm-pack/releases/download/v0.10.2/wasm-pack-v0.10.2-x86_64-pc-windows-msvc.tar.gz" | \
  tar -xz
```

The above outputs:

```
tar: Skipping to next header
tar: A lone zero block at 12975
tar: Exiting with failure status due to previous errors
```

#### 🤔 Expected Behavior

The tar should be able to be extracted like v0.10.1

```bash
curl -L "https://github.com/rustwasm/wasm-pack/releases/download/v0.10.1/wasm-pack-v0.10.1-x86_64-pc-windows-msvc.tar.gz" | \
  tar -xz
```

#### Workaround

Current workaround to downgrade to 0.10.1:

```diff
-curl https://rustwasm.github.io/wasm-pack/installer/init.sh -sSf | sh 
+npm install -g wasm-pack@0.10.1
```

**Comment by tmcguire** [comment]
This is causing our GitHub Actions CI to fail on Windows with the following:

```
/c/Rust/.cargo/bin/wasm-pack: cannot execute binary file: Exec format error
```

Just putting it here in case someone searches for that error message.

Downgrading wasm-pack to 0.10.1 fixed our CI problem.

**Comment by nickbabcock** [comment]
Current workaround to downgrade to 0.10.1 (OP has been updated to include this):

```diff
-curl https://rustwasm.github.io/wasm-pack/installer/init.sh -sSf | sh 
+npm install -g wasm-pack@0.10.1
```

As for the root cause, maybe there is a bug in 7zip? I see it is used to [create the tarball](https://github.com/rustwasm/wasm-pack/runs/4552304830?check_suite_focus=true#step:13:2), and verified reported size matches download size. 7zip can extract the generated tarball, but it prints a warning, so if 0.10.2 is necessary, one can manually download the windows tarball and extract with 7zip while ignoring warnings.

**Comment by magcius** [comment]
Would it be possible to just push out a new release to see if the bug happens again? How can we help solve this going forward?

**Comment by Keavon** [comment]
@drager just checking if there's a possible timeline to this getting resolved. We're having to pin to the previous version right now, but that old one is broken for Apple Silicon users.

**Comment by RReverser** [comment]
@drager Ping. This is a pretty significant issue that's blocking usage of latest wasm-pack on Windows, and was open for 5 months. Any chance of making another release?

**Comment by drager** [comment]
> @drager Ping. This is a pretty significant issue that's blocking usage of latest wasm-pack on Windows, and was open for 5 months. Any chance of making another release?

Sorry for my late response and acting here. Got limited time recently... I will take a look this weekend and try to create a new release!

**Comment by RReverser** [comment]
> > @drager Ping. This is a pretty significant issue that's blocking usage of latest wasm-pack on Windows, and was open for 5 months. Any chance of making another release?
> 
> Sorry for my late response and acting here. Got limited time recently... I will take a look this weekend and try to create a new release!

@drager I'm guessing it didn't work out 😅

**Comment by drager** [comment]
> > > @drager Ping. This is a pretty significant issue that's blocking usage of latest wasm-pack on Windows, and was open for 5 months. Any chance of making another release?
> > 
> > 
> > Sorry for my late response and acting here. Got limited time recently... I will take a look this weekend and try to create a new release!
> 
> @drager I'm guessing it didn't work out 😅

No, sorry 😞 Will take a look soon!

**Comment by ConnorJamesLow** [comment]
Hi, can confirm this is still an issue :)

**Comment by nasso** [comment]
I just opened a pull request with a very simple fix: #1144

**Comment by drager** [comment]
Released in [0.10.3](https://crates.io/crates/wasm-pack/0.10.3).

**Comment by Keavon** [comment]
Thanks @drager! And just checking, these failing CI tests are not a problem?
